caikit/runtime/client/remote_config.py

```diff
@@ -0,0 +1,210 @@
+
+            model_path (str):
+                The path used to load this module
+
+            connection_info ConnectionInfo:
+                The connection information of the remote to use
+
+            protocol: str
+                The protocol to connect with
+
+            model_key: str
+                The model key to use when sending GRPC requests. An example is mm-model-id
+
+           Returns:
+               model_config (RemoteModuleConfig): Instantiated RemoteModuleConfig for
+               model given model_path.
+        """
+        # Validate model path arg
+        error.type_check("<COR71170339E>", str, model_path=model_path)
+
+        # Get local module reference
+        error.type_check(
+            "<COR71270339E>",
+            str,
+            ModuleBase,
+            _ModuleBaseMeta,
+            module_reference=module_reference,
+        )
+        if isinstance(module_reference, ModuleBase) or (
+            inspect.isclass(module_reference)
+            and issubclass(module_reference, ModuleBase)
+        ):
+            local_module_class = module_reference
+        else:
+            if module_reference not in module_registry():
+                raise KeyError(f"Unknown module reference {module_reference}")
+
+            local_module_class = module_registry().get(module_reference)
+
+        # Construct model config dict
+        remote_config_dict = {
+            # Connection info
+            "connection": connection_info,
+            "protocol": protocol,
+            "model_key": model_key,
+            # Method info
+            "task_methods": [],
+            "train_method": None,
+            # Source module info
+            "model_path": model_path,
+            "module_id": f"{local_module_class.MODULE_ID}-remote",
+            "module_name": f"{local_module_class.MODULE_NAME} Remote",
+        }
+
+        # Parse inference methods signatures
+        for task_class in local_module_class.tasks:
+            task_methods = []
+            for input, output, signature in local_module_class.get_inference_signatures(
+                task_class
+            ):
+
+                # Don't get the actual DataBaseObject as the ServicePackage might not have
+                # been generated
+                request_class_name = get_task_predict_request_name(
+                    task_class, input, output
+                )
+                task_request_name = get_task_predict_rpc_name(task_class, input, output)
+
+                if hasattr(signature.return_type, "__name__"):
+                    task_return_type = signature.return_type.__name__
+                else:
+                    task_return_type = get_origin(signature.return_type).__name__
+
+                # Get the underlying DataBaseObject for stream types
+                if output and get_args(signature.return_type) != ():
+                    task_return_type = get_args(signature.return_type)[0].__name__
```

Comment:
This gets what type of DataStream the return type is. Since the format will always be `DataStream[<generic>]` we can use `[0]` to get the `<generic>` type.